zerver/views/video_calls.py

```diff
@@ -185,6 +186,57 @@ def make_user_authenticated_zoom_video_call(
     return json_success(request, data={"url": res.json()["join_url"]})
 
 
+def get_zoom_server_to_server_access_token() -> str:
```

Comment:
These server access tokens last an hour, so we could store them in a field on the realm and check if it's expired before generating a new one. We'd need to calculate the time we requested the token as the Zoom payload gives how long until the token expires, but not the time is was generated.

Example payload from [Zoom developer docs](https://developers.zoom.us/docs/internal-apps/s2s-oauth/#to-generate-an-access-token):
```json
{
      "access_token": "eyJhbGciOiJIUzUxMiIsInYiOiIyLjAiLCJraWQiOiI8S0lEPiJ...",
      "token_type": "bearer",
      "expires_in": 3599,
      "scope" : "user:read:admin",
      "api_url": "https://api.zoom.us"
}
```

Reply:
I don't want to store an additional field on Realm; it's expensive to make that table bigger. But this would be a reasonable data structure to store in `redis` or memcached; that's a good place for transient structures valuable on that time scale.

Reply:
(If we do that, we'll need to be careful to make sure we model `expires` correctly -- converting it into an expiration time and probably cutting 2 minutes because several network round-trip delays are involved and we want to just get a new one if it's almost expired).